src/Castle.Windsor.Extensions.MsDependencyInjection/ServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,79 @@
+// Copyright 2004-2020 Castle Project - http://www.castleproject.org/
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace Castle.Windsor.Extensions.MsDependencyInjection
+{
+    using System;
+    using System.Reflection;
+
+    using Castle.MicroKernel.Registration;
+    using Castle.Windsor;
+    
+    using Microsoft.Extensions.DependencyInjection;
+
+    internal static class ServiceCollectionExtensions
+    {
+        public static IWindsorContainer CreateContainer(this IServiceCollection serviceCollection, WindsorServiceProviderFactory factory)
```

Comment:
Does it give meaning to have a method where we also pass the container? Applications might want items preregistered before added the "ServiceCollection" they might also want control of the instance in there bootstrapper.?

Reply:
Not the way .NET Core works. You should not create your own container. 
`ConfigureServices` defines the service collection. This in turns calls `WindsorServiceProviderFactory.CreateBuilder` which should create new container every time is called. If you want register components / services directly with Windsor Container you should implement implement `void ConfigureContainer(container)` in `Startup`

Reply:
When Gaven did the aspnetcore facility we had the same discussion. Also about what should be public and/or virtual or not. He wanted it closed also.
After the aspnetcore was release for 5.0.0 a month or two later, people asked for it to be opened..

I don’t see why we should not have a method also allowing to pass a IWindsorContainer. I don’t like things locking me to definitive instances.

Please add a method where you can pass the IWindsor Container. Or add it as a parameter with a default value, and if null, new it in the create...